src/cmap/commands.ts

```diff
@@ -851,8 +858,9 @@ export class BinMsg {
       if (payloadType === 0) {
         const bsonSize = this.data.readUInt32LE(this.index);
         const bin = this.data.slice(this.index, this.index + bsonSize);
-        this.documents.push(raw ? bin : BSON.deserialize(bin, _options));
-
+        this.documents.push(
+          raw ? bin : BSON.deserialize(bin, Object.assign({ validation }, _options))
```

Comment:
instead of doing this in two places - here and in L877, can we instead add it to the BSONSerializeOptions `_options` variable above? And to avoid touching the `BSONSerializeOptions` type, you can probably do `_options: BSONSerializeOptions & { validation: <type desrciption> }`

Reply:
done!